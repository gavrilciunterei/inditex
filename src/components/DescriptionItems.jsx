import React from 'react';
import TextDescription from './TextDescription';

function DescriptionItems({ name, text }) {
  return (
    <div>
      <div className="inline-flex">
        <TextDescription text={name} />
        <p>{typeof text === 'object' ? text.join(', ') : text}</p>
      </div>
    </div>
  );
}

export default DescriptionItems;
