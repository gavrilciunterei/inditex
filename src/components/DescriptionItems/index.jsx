import React from 'react';
import TextDescription from '../TextDescription';

function DescriptionItems({ name, text }) {
  return (
    <div>
      <div className="inline-flex">
        <TextDescription text={name} />
        <p>
          {typeof text === 'object'
            ? text.join(', ')
            : text.length < 1
            ? '?'
            : text}
        </p>
      </div>
    </div>
  );
}

export default DescriptionItems;
