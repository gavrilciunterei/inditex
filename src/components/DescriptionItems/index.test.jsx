import React from 'react';
import renderer from 'react-test-renderer';
import DescriptionItems from './index';

it('Test DescriptionItems', () => {
  const tree = renderer
    .create(<DescriptionItems name="Test " text="DescriptionItems" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test DescriptionItems with object', () => {
  const tree = renderer
    .create(<DescriptionItems name="Prueba" text={['test', 'object']} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
