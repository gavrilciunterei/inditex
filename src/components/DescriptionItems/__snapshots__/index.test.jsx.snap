// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DescriptionItems 1`] = `
<div>
  <div
    className="inline-flex"
  >
    <p
      className="font-bold mr-2"
    >
      Test 
      :
    </p>
    <p>
      DescriptionItems
    </p>
  </div>
</div>
`;

exports[`Test DescriptionItems with object 1`] = `
<div>
  <div
    className="inline-flex"
  >
    <p
      className="font-bold mr-2"
    >
      Prueba
      :
    </p>
    <p>
      test, object
    </p>
  </div>
</div>
`;
